rfc/rfc-59/rfc-59.md

```diff
@@ -0,0 +1,285 @@
+
+  @Override
+  public Option<Map<String, String>> getMetadata() ;
+}
+
+```
+
+### Deduplicate record with multiple event-time fields
+
+This function will deduplicate records with same primay key(s) by multiple event-time fields and only keep record with latest values in every event-time field.
+Since Hudi buffers records in a buckect before flushing or checkpointing, so it's very necessary to conduct deduplication inside a bucket in order to sink only one record (whose every event-time is latest) to next step where this record will compare with record on Disk to decide which one should be remained or should some part of the persisted record be updated.
+
+```java
+// hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java
+public class FlinkWriteHelper<T extends HoodieRecordPayload, R> extends AbstractWriteHelper<T, List<HoodieRecord<T>>, List<HoodieKey>, List<WriteStatus>, R> {
+  
+  // ...
+
+  /**
+  *
+  * This function deduplicates records having same primary keys by multiple event-time fields. The only record with the latest event-time fields will be remained.
+  * It's very like another function (FlinkWriteHelper # deduplicateRecords) in this class. The difference is that this function use multiple event-time to deduplicate records.
+  */
+  public List<HoodieRecord<T>> deduplicateRecordsWithMultipleCombineFields(List<HoodieRecord<T>> records, HoodieIndex<T, ?, ?, ?> index, int parallelism) ;
+
+  // ...
+}
+```
+
+### Provide more Configurations to improve usability
+
+More importantly, we should create options to let developers easily config which fields are event-time fields that need to be conduct latest verification.
+
+Firstly, we should add a config in `HoodiePayloadProps`.
+
+```java
+// hudi-common/src/main/java/org/apache/hudi/common/model/HoodiePayloadProps.java
+public class HoodiePayloadProps {
+
+  // ...
+
+  /**
+   * Newly added config to support multiple event-time latest verification fields instead of only one 'ts'
+   * The configuration would look like below when creating Hudi Table:
+   * 'hoodie.eventime.latest.verification.fields' = 'event_a_ts,event_b_ts,event_c_ts'
+   * */
+  public static final String PAYLOAD_COMBINE_DEPENDING_ON_FIELDS = "hoodie.eventime.latest.verification.fields";
+
+  // ...
+}
+```
+
+Secondly, we should add an option in `FlinkOptions`.
+
+```java
+// hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java
+public class FlinkOptions extends HoodieConfig {
+  // ...
+  // ------------------------------------------------------------------------
+  //  Multiple event-time fields latest verification related options
+  // ------------------------------------------------------------------------
+
+  public static final ConfigOption<Boolean> IS_USING_MULTIPLE_COMBINE_FIELDS = ConfigOptions
+      .key("multiple.eventime.latest.verification.enable")

```

Comment:
we could avoid adding this new field if we automatically deduce single vs multiple ordering fields